src/main/java/lotto/domain/winning/WinningNumbers.java

```diff
@@ -5,30 +5,41 @@
 import lotto.domain.grade.Grade;
 import lotto.domain.lotto.Lotto;
 import lotto.domain.lotto.LottoNumber;
+import lotto.error.exception.SizeExceedLottoException;
 
 public class WinningNumbers {
 
     private final List<LottoNumber> winningNumbers;
 
-    public WinningNumbers(List<LottoNumber> winningNumbers) {
+    private final LottoNumber bonusNumber;
+
+    public WinningNumbers(List<LottoNumber> winningNumbers, LottoNumber bonusNumber) {
         this.winningNumbers = winningNumbers;
+        this.bonusNumber = bonusNumber;
     }
 
-    public static WinningNumbers fromValues(final List<Integer> values) {
+    public static WinningNumbers fromValues(final List<Integer> values, LottoNumber bonusNumber) {
+        if (values.size() != 6) {
+            throw new SizeExceedLottoException(values.size());
+        }
+
         List<LottoNumber> lottoNumbers = values.stream()
             .map(LottoNumber::fromInt)
             .collect(Collectors.toList());
-        return new WinningNumbers(lottoNumbers);
+        return new WinningNumbers(lottoNumbers, bonusNumber);
     }
 
     public Grade confirmWinning(final Lotto lotto) {
-        return Grade.fromCorrectingCount(findNumberThatMatchWinningBall(lotto));
+        return Grade.fromMatchResult(findMatchResultByLotto(lotto));
+    }
+
+    private WinningNumberResult findMatchResultByLotto(final Lotto lotto) {
+        return new WinningNumberResult(calculateMatchingLottoNumbers(lotto),
+            lotto.contains(bonusNumber));
     }
 
-    private int findNumberThatMatchWinningBall(final Lotto lotto) {
-        return (int) winningNumbers.stream()
-            .filter(lotto::contains)
-            .count();
+    private int calculateMatchingLottoNumbers(Lotto lotto) {
+        return (int) winningNumbers.stream().filter(lotto::contains).count();
```

Comment:
다음 객체지향 생활 체조 원칙을 참고해주세요.
> 규칙 4: 한 줄에 점을 하나만 찍는다.

```suggestion
        return (int) winningNumbers.stream()
            .filter(lotto::contains)
            .count();
```

Reply:
선동님 혹시 인텔리제이 자동 설정을 어떻게 사용하시는지 알 수 있을까요?
cmd + option + l 누르면 자동 줄 설정이 되는데 한 번 더누르니까 한줄로 만들어버리는거 같네요 ㅠㅠ

Reply:
저는 따로 코드 포맷팅을 편집해서 사용하진 않고 `IntelliJ IDEA`에서 제공하는 기본 포맷팅을 사용 중입니다.
stream을 사용할 땐 어느 정도 습관이 배서 그런지 줄바꿈 처리를 직접 하고 있습니다.

Reply:
저도 습관을 들여야겠네요 답변 감사드립니다!